recipes/librdkafka/all/conanfile.py

```diff
@@ -66,60 +67,71 @@ def requirements(self):
             self.requires("zstd/1.5.2")
         if self.options.ssl:
             self.requires("openssl/1.1.1q")
-        if self.options.sasl and self.settings.os != "Windows":
+        if self._depends_on_cyrus_sasl:
             self.requires("cyrus-sasl/2.1.27")
         if self.options.get_safe("curl", False):
             self.requires("libcurl/7.84.0")
 
     def build_requirements(self):
-        if self.options.sasl and self.settings.os != "Windows":
-            self.build_requires("pkgconf/1.7.4")
+        if self._depends_on_cyrus_sasl:
+            self.tool_requires("pkgconf/1.7.4")
 
-    def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+    def layout(self):
+        cmake_layout(self)
+        self.folders.source = "src"
+        self.folders.build = "build"
```

Comment:
These should be the defaults 🤔

Reply:
Good question

signature of cmake_layout is `cmake_layout(conanfile, generator=None, src_folder=".")`

So the default layout of `cmake_layout`:

```
build/
  <package_id>/
    source_code_files
    patches/
    build/<build_type>
source/
    source_code
    patches/
```

- source code and patches folder are mixed up, as well as build folder and source code (not a fully out of source build).
- and less important: `build/<build_type>` is unnecessarily long (might lead to path length issue on Windows)

So I propose instead this layout:

```
build/
  <package_id>/
    build/
    patches/
    src/source_code_files
source/
    patches/
    src/source_code_files
```